myapp/static/js/myapp/single_stock.js

```diff
@@ -137,11 +141,20 @@ function getRandomRgba() {
 }
 }
 
-function compareTwoStocks(originSymbol) {
+function addStockToCompare() {
     var stockNameOption = document.forms["compareForm"]['stockNameToCompare']['value'];
-    if (stockNameOption != '') {
-        secondarySymbol = stockNameOption.slice(0, stockNameOption.indexOf(','))
-        getHistoricData(originSymbol,recent_time_range_chose ,secondarySymbol);
-        recent_symbol_to_compare = secondarySymbol;
+    secondarySymbol = stockNameOption.slice(0, stockNameOption.indexOf(','))
+    if (stockNameOption != '' && secondarySymbol != originSymbol && !recent_symbols_to_compare.some(value => secondarySymbol == value)) {
```

Comment:
Excuse my ignorance but what does `!recent_symbols_to_compare.some(value => secondarySymbol == value))` exactly test for in the array? 🤔

Reply:
this test would prevent duplicated stocks graphs, so whenever the use click twice on compare button, only one would be added